packages/connectors/src/metaMask.ts

```diff
@@ -283,11 +283,17 @@ export function metaMask(parameters: MetaMaskParameters = {}) {
               rpcUrls,
             } satisfies AddEthereumChainParameter
 
-            await provider.request({
-              method: 'wallet_addEthereumChain',
-              params: [addEthereumChain],
-            })
-
+            await Promise.all([
+              provider.request({
+                method: 'wallet_addEthereumChain',
+                params: [addEthereumChain],
+              }),
```

Comment:
Should this also have the additional have the additional then block to fire the change event if its needed?

https://github.com/wevm/wagmi/blob/ceadbbe99b987149aeccc9411897431e4c7f0ab2/packages/connectors/src/metaMask.ts#L230-L239